src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -154,18 +163,23 @@ public void visitToken(final DetailAST ast) {
     @Override
     public void leaveToken(DetailAST ast) {
         final int astType = ast.getType();
-        if (astType != TokenTypes.VARIABLE_DEF
+        if (astType == TokenTypes.SLIST) {
+            leaveSlist(ast);
+        }
+        else if (astType == TokenTypes.LITERAL_NEW) {
+            leaveLiteralNew(ast);
```

Comment:
This is again just a re-ordering of old code and moving it to a separate method.